packages/flame/lib/src/sprite_batch.dart

```diff
@@ -172,28 +180,72 @@ class SpriteBatch {
     Images? images,
     bool useAtlas = true,
   }) async {
+    final _images = images ?? Flame.images;
+    //flippableAtlas = await _atlasFromCache(_images, path);
     return SpriteBatch(
-      await _generateAtlas(images, path),
+      await _images.load(path),
       defaultColor: defaultColor,
       defaultTransform: defaultTransform ?? RSTransform(1, 0, 0, 0),
       defaultBlendMode: defaultBlendMode,
       useAtlas: useAtlas,
+      flippableAtlas: await _images.fetchOrGenerate(
```

Comment:
If you move this to where the field is declared it should be lazily initialized, now it'll be initialized directly.